hal/halx86/xbox/reboot.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * PROJECT:         Xbox HAL
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Xbox reboot functions
+ * COPYRIGHT:       Copyright 2004 Lehner Franz (franz@caos.at)
+ *                  Copyright 2019 Stanislav Motylkov (x86corez@gmail.com)
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include "halxbox.h"
+
+#define NDEBUG
+#include <debug.h>
+
+/* PRIVATE FUNCTIONS *********************************************************/
+
+VOID
+NTAPI
+SMBusWriteByte(UCHAR Address, UCHAR Register, UCHAR Data)
+{
+    INT Retries = 50;
+
+    /* Wait while bus is busy with any master traffic */
+    while (READ_PORT_USHORT((PUSHORT)SMB_GLOBAL_STATUS) & 0x0800);
+
+    while (Retries--)
+    {
+        UCHAR b;
+
+        WRITE_PORT_UCHAR((PUCHAR)SMB_HOST_ADDRESS, Address << 1);
+        WRITE_PORT_UCHAR((PUCHAR)SMB_HOST_COMMAND, Register);
+
+        WRITE_PORT_UCHAR((PUCHAR)SMB_HOST_DATA, Data);
+
+        /* Clear down all preexisting errors */
+        WRITE_PORT_USHORT((PUSHORT)SMB_GLOBAL_STATUS, READ_PORT_USHORT((PUSHORT)SMB_GLOBAL_STATUS));
+
+        /* Let I2C SMBus know we're sending a single byte here */
+        WRITE_PORT_UCHAR((PUCHAR)SMB_GLOBAL_ENABLE, 0x1A);
+
+        b = 0;
+
+        while (!(b & 0x36))
```

Comment:
Any of these 4 bits can be set?

Reply:
Probably yes, at least I can confirm this code works. :slightly_smiling_face: